katana/native/cases/static/cases/js/casesApp.js

```diff
@@ -0,0 +1,1040 @@
+ 							popup.find("#StepKeyword").append($('<option>',{ value: x,  text: x }));
+ 						}
+ 					popup.find('#StepKeyword').val(oneCaseStep["@Keyword"]);
+				
+					jQuery.getJSON("./cases/getListOfComments/?driver="+driver+"&keyword="+keyword).done(function(data) {
+	 					a_items = data['fields'];
+		 				console.log("received", data, a_items);
+
+		 				out_array = a_items[0]['comment'];
+		 				var outstr = out_array.join("<br>");
+		 				console.log(outstr);
+
+		 				caseApp.lastPopup.find("#sourceCaseFileText").html(""); 
+		 				caseApp.lastPopup.find("#sourceCaseFileText").html(outstr);
+
+					});
+			});
+		});
+		//console.log(xdata);
+		console.log(oneCaseStep);
+		popup.find("#StepRowToEdit").attr("value",sid);
+
+		//popup.find("#StepDriver").attr("value",oneCaseStep[ "@Driver"]);
+		popup.find("#StepDriver").val(oneCaseStep[ "@Driver"]);
+		console.log(popup.find("#StepDriver").val());
+		//popup.find("#StepKeyword").attr("value",oneCaseStep[ "@Keyword"]);
+		popup.find("#StepKeyword").val(oneCaseStep[ "@Keyword"]);
+
+		
+		//alert("Keyword = "+keyword+" driver = "+ driver + " values = "+ popup.find("#StepDriver").val(oneCaseStep[ "@Driver"]));
+		popup.find("#StepTS").attr("value",oneCaseStep["@TS"]);
+		popup.find("#StepDescription").attr("value",oneCaseStep["Description"]);
+		popup.find("#StepContext").attr("value",oneCaseStep["context"]);
+		popup.find("#SteponError-at-action").attr("value",oneCaseStep['onError']["@action"]);
+		popup.find("#SteponError-at-value").attr("value",oneCaseStep['onError']["@value"]);
+		popup.find("#runmode-at-type").attr("type",oneCaseStep["runmode"]["@type"]);
+		popup.find("#runmode-at-value").attr("value",oneCaseStep["runmode"]["@value"]);
+		popup.find("#StepImpact").attr("value",oneCaseStep["impact"]);
+		popup.find("#StepInputDataFile").attr("value",oneCaseStep["InputDataFile"]);
+		popup.find('.rule-condition').hide();
+		if (oneCaseStep["Execute"]['@ExecType']) {
+			console.log("FOUND EXECT TYPE ",oneCaseStep["Execute"]['@ExecType'] )
+			if (oneCaseStep["Execute"]['@ExecType'] == 'If' || oneCaseStep["Execute"]['@ExecType'] == 'If Not') {
+				popup.find('.rule-condition').show();
+			}
+			
+		}
+		if (oneCaseStep["Execute"]['Rule']) {
+			popup.find('#executeRuleAtCondition').attr('value',oneCaseStep["Execute"]['Rule']['@Condition']);
+			popup.find('#executeRuleAtCondvalue').attr('value',oneCaseStep["Execute"]['Rule']['@Condvalue']);
+			popup.find('#executeRuleAtElse').attr('value',oneCaseStep["Execute"]['Rule']['@Else']);
+			popup.find('#executeRuleAtElsevalue').attr('value',oneCaseStep["Execute"]['Rule']['@Elsevalue']);
+		}
+	//katana.popupController.updateActiveWindow(popup);
+
+	caseApp.makePopupArguments(popup, oneCaseStep);
+
+	// Now  set the callbacks once the DOM has new HTML elements in it
+	
+	// Fill in the value based on keyword and action 
+	var opts = jQuery.getJSON("./cases/getListOfComments/?driver="+driver+"&keyword="+keyword).done(function(data) {
```

Comment:
Remove line 621 because kws and associated data will be fetched together

Reply:
Done

Reply:
Added signature arguments.